### ISSUE TITLE
"Could not parse remote_state block from target config" when expressing dependency as of v0.28.0

### ISSUE THREAD

**Issue by jtackaberry** [gruntwork-io/terragrunt#1625]
I have a test project with the following structure:

```
├── lab
│   ├── nsg
│   │   └── terragrunt.hcl
│   ├── tfstate
│   │   └── terragrunt.hcl
│   ├── terragrunt.hcl
│   └── tfvars.hcl
└── modules
    ├── nsg
    │   └── main.tf
    └── tfstate
        └── main.tf
```

The project just creates an Azure blog storage container to hold tfstate, and an Azure Network Security Group, both in an existing subscription and resource group.

The nsg config declares a dependency on the tfstate config, to ensure appropriate ordering with run-all.  `lab/nsg/terragrunt.hcl` contains:

```
dependency "tfstate" {
  config_path = "${get_terragrunt_dir()}/../tfstate"
}
```

However, with this stanza, I get the following warning on running terragrunt commands against the nsg config:

```
WARN[0000] Could not parse remote_state block from target config /home/jtackaberry/tmp/tgtest/lab/tfstate/terragrunt.hcl
WARN[0000] Falling back to terragrunt output.
```

Indeed, this file doesn't contain a `remote_state` block.  That's defined in `lab/terragrunt.hcl`, which is included by the `nsg` config (and would be included in general by all non-`tfstate` config).

Near as I can tell, this warning is purely cosmetic -- everything seems to function otherwise.  It also didn't occur prior to v0.28.0.  The warning goes away if I remove the dependency block quoted above, but then of course `terragrunt run-all apply` wouldn't about the dependency and could apply the config in the wrong order.

Is this a bug, or am I expressing the dependency in the wrong way?

I am including the full `tgtest` project as an attachment here.  Thanks!

[tgtest.tar.gz](https://github.com/gruntwork-io/terragrunt/files/6240194/tgtest.tar.gz)

**Comment by yorinasub17** [comment]
Apologies for the confusion here!

Based on what I see, it does look like you have everything configured correctly.

That warning is indeed benign and safe to ignore. It is indicating that the default dependency fetching optimization can not be done because terragrunt didn't detect the `remote_state` block. We initially set it at warning level so it is more visible, but looking back at it now, it probably should be info logs.

You can refer to the section "Can I speed up dependency fetching?" in [the `dependency` block documentation for more details](https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#dependency).

**Comment by yorinasub17** [comment]
FYI the warning is now a debug log message. Will close this as resolved, but let me know if you have further questions about what I mentioned above!

**Comment by jtackaberry** [comment]
@yorinasub17 many thanks.

**Comment by s1mark** [comment]
I have tried 0.28.18 and still get this warning

**Comment by yorinasub17** [comment]
This has not been released yet, and will go out as `0.28.19` tomorrow.

**Comment by yorinasub17** [comment]
FYI [0.28.19](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.28.19) is now available with this change.
